Src/DependencyCollector/Shared/ServiceBusDiagnosticsEventHandler.cs

```diff
@@ -0,0 +1,171 @@
+                    this.OnRequest(evnt.Key, evnt.Value, currentActivity);
+                    break;
+                case "Microsoft.Azure.ServiceBus.Exception":
+                    break;
+                default:
+                    this.OnDependency(evnt.Key, evnt.Value, currentActivity);
+                    break;
+            }
+        }
+
+        private void OnDependency(string name, object payload, Activity activity)
+        {
+            DependencyTelemetry telemetry = new DependencyTelemetry { Type = RemoteDependencyConstants.AzureServiceBus };
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+            
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            string endpoint = this.FetchPayloadProperty<Uri>(name, EndpointPropertyName, payload)?.ToString();
+
+            // Queue/Topic name, e.g. myqueue/mytopic
+            string queueName = this.FetchPayloadProperty<string>(name, EntityPropertyName, payload);
+
+            // Target uniquely identifies the resource, we use both: queueName and endpoint 
+            // with schema used for SQL-dependencies
+            telemetry.Target = string.Join(" | ", queueName, endpoint);
+            this.telemetryClient.TrackDependency(telemetry);
+        }
+
+        private void OnRequest(string name, object payload, Activity activity)
+        {
+            RequestTelemetry telemetry = new RequestTelemetry();
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+
+            string endpoint = this.FetchPayloadProperty<Uri>(name, EndpointPropertyName, payload)?.ToString();
+
+            // Queue/Topic name, e.g. myqueue/mytopic
+            string queueName = this.FetchPayloadProperty<string>(name, EntityPropertyName, payload);
+
+            // We want to make Source field extendable at the beginning as we may add
+            // multi ikey support and also build some special UI for requests coming from the queues
+            // it follows Request.Source schema invented for multi ikey
+            telemetry.Source = string.Format(CultureInfo.InvariantCulture, "type:{0} | name:{1} | endpoint:{2}", RemoteDependencyConstants.AzureServiceBus, queueName, endpoint);
+
+            this.telemetryClient.TrackRequest(telemetry);
+        }
+
+        private void SetCommonProperties(string eventName, object eventPayload, Activity activity, OperationTelemetry telemetry)
+        {
+            // activity name looks like 'Microsoft.Azure.ServiceBus.<Name>'
+            // as namespace is too verbose, we'll just take the last node from the activity name as telemetry name
+            string activityName = activity.OperationName;
+            int lastDotIndex = activityName.LastIndexOf('.');
+            if (lastDotIndex >= 0)
+            {
+                telemetry.Name = activityName.Substring(lastDotIndex + 1);
```

Comment:
So I'm not sure if this *Name* will work equally well for __Dependencies__ and __Requests__ - Dependencies have the *Type* property to combine it with but Requests don't and the *Source* field does not seem to be used in traditional UX. So this way "Process" Requests from SB and EH will be jammed together under same UX entry. Maybe we should use sth like ```{OperationName} from {DependencyTypeName}``` just with the proper processing - so in this case it would be ```Process from Azure ServiceBus```. Thoughts?

Reply:
If we continue HTTP analogy, Name describes king of operation that happens on this node (i.e. GET controller).
Here we have a verb only (Process) and no route (controller).

While Source describes where to operation came from. I believe type belongs to the Source (until it has a special field).

Also, this part never ever changes. What is the benefit of including it into the telemetry?